### PR TITLE
Fixes chest hits gibbing the player

### LINKED ISSUE #7351: You can dismember torsos, causing immediate death and other bugs.
### Environment
- Game Version: Staging 21090215
- OS Version: Windows 10

### Detailed description of bug
Using weapons such as energy swords or chainsaws, you can dismember body parts when attacking them. This includes the torso of players, which when dismembered not only immediately kills the players in question but also severs every single one of their body parts without dropping any of their worn items. Torsos should not be able to be dismembered like this.

### Steps to reproduce
1. Obtain a traitor's energy sword.
2. Attack someone's torso until you dismember it.
3. Witness that someone collapse into a pile of limbs without dropping any of their items.

### PR DESCRIPTION
**Pull request by Aranclanos**
### Purpose
CL: [Fix] Fixed brute hits on chest gibbing all bodyparts.
Fixes #7351
Fixes bodyparts chance almost always being above 100%
Changes gibbing chance and armor values from a range of `0 to 1` to `0 to 100`

### CODE REVIEW

**Comment by PerfectTangent** [comment]
Why should armor start at a minimum of 1?

**Comment by Aranclanos** [comment]
typo, its 0 to 100

**Comment by PerfectTangent** [comment]
So negative armor is not possible anymore?

**Comment by Aranclanos** [comment]
wasnt possible before this pr, at least on dismemberment armor

**Comment by PerfectTangent** [comment]
Ah, dismemberment armor. Got it.

File: UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartTraumaDamage.cs [hunk 1]
```diff
@@ -439,10 +439,16 @@ public void InternalBleedingLogic(AttackType attackType = AttackType.Internal, D
...
+			if (GibChance == 0)
+				return;
+			var armorChanceModifer = GibChance - SelfArmor.DismembermentProtectionChance;
+			if (Severity == DamageSeverity.Max || currentCutSize == BodyPartCutSize.LARGE)
+			{
+				armorChanceModifer += 25; //Make it more likely that the bodypart can be gibbed in it's worst condition.
+			}
+			
+			var chance = UnityEngine.Random.RandomRange(0, 100);
+			if(chance < armorChanceModifer)
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartTraumaDamage.cs]:
```suggestion
			var chance = UnityEngine.Random.Range(0, 101);
			if(chance < armorChanceModifer)
```
RandomRange is obsolete and Range for ints is exclusive for max value.

**Comment by Aranclanos** [comment]
Ran that chance a million times to make sure it wasnt off

**Comment by NoooneyDude** [comment]
> Ran that chance a million times to make sure it wasnt off

So it was fine to not include `100`?

**Review by Bod9001 — APPROVED** [review]

**Review by ThatDan123 — APPROVED** [review]

**Pull request merged**